### PR TITLE
[Nifty-A Clock] ClockFace

### PR DESCRIPTION
**Pull request by alessandrococco**
This PR upgrades the **Nifty-A Clock** to the  **ClockFace** library (thank @rigrig!)

(I plan to update **Nifty-B Clock** and some other clocks as well)

### CODE REVIEW

File: apps/ffcniftya/app.js [hunk 1]
```diff
@@ -30,77 +11,50 @@ function ISO8601_week_no(date) {
...
-  const centerDatesScaleX = center.x + 40 * scale;
-  g.setFontAlign(-1, 0).setFont("Vector", 16 * scale);
-  g.drawString(year, centerDatesScaleX, center.y - 62 * scale);
-  g.drawString(month, centerDatesScaleX, center.y - 44 * scale);
-  g.drawString(day, centerDatesScaleX, center.y - 26 * scale);
-  if (showWeekNum) g.drawString(weekNum, centerDatesScaleX, center.y + 15 * scale);
-  g.drawString(monthName, centerDatesScaleX, center.y + 48 * scale);
-  g.drawString(dayName, centerDatesScaleX, center.y + 66 * scale);
-}
+    this.showWeekNum = Object.assign({ showWeekNum: true }, require("Storage").readJSON("ffcniftya.json", true))["showWeekNum"];
```
Comment (rigrig) [inline apps/ffcniftya/app.js:37]: I don't think this does what you want it to do: it always sets `this.showWeekNum` to an object, so `if (this.showWeekNum)` below will always evaluate to true.

What you could do is
* specify `settingsFile: "ffcniftya.json",` to have ClockFace read in the settings for you
* check for `if (this.showWeekNum !== false)` below, (making it default to enabled if not set).

File: apps/ffcniftya/app.js [hunk 2]
```diff
@@ -30,77 +11,50 @@ function ISO8601_week_no(date) {
...
-Bangle.on('lcdPower', (on) => {
-  if (on) {
-    tick();
-  } else {
-    clearTickTimer();
+    g.setFontAlign(-1, 0).setFont("Vector", 16 * this.scale);
+    g.drawString(date.getFullYear(date), this.centerDatesScaleX, this.center.y - 62 * this.scale);
+    g.drawString(format(month), this.centerDatesScaleX, this.center.y - 44 * this.scale);
+    g.drawString(format(date.getDate()), this.centerDatesScaleX, this.center.y - 26 * this.scale);
+    if (this.showWeekNum) g.drawString(format(ISO8601_week_no(date)), this.centerDatesScaleX, this.center.y + 15 * this.scale);
```
Comment (rigrig) [inline apps/ffcniftya/app.js:55]:
```suggestion
    if (this.showWeekNum !== false) g.drawString(format(ISO8601_week_no(date)), this.centerDatesScaleX, this.center.y + 15 * this.scale);
```

File: apps/ffcniftya/app.js [hunk 3]
```diff
@@ -30,77 +11,50 @@ function ISO8601_week_no(date) {
...
+    g.drawString(monthName, this.centerDatesScaleX, this.center.y + 48 * this.scale);
+    g.drawString(dayName, this.centerDatesScaleX, this.center.y + 66 * this.scale);
   }
```
Comment (rigrig) [inline apps/ffcniftya/app.js:58]:
```suggestion
  },
  settingsFile: "ffcniftya.json"
```

File: apps/ffcniftya/app.js (hunk 1 above)
Comment (alessandrococco) [reply apps/ffcniftya/app.js:37]: > I don't think this does what you want it to do: it always sets this.showWeekNum to an object, so if (this.showWeekNum) below will always evaluate to true.

I just tried to change the setting and it works. Maybe you didn't notice the ..["showWeekNum"] at the end?

Comment (rigrig) [reply apps/ffcniftya/app.js:37]: Whoops, you're right, sorry.

Comment (alessandrococco) [reply apps/ffcniftya/app.js:37]: No problem! 😄

**Comment by gfwilliams** [comment]
Did you want to add `settingsFile: "ffcniftya.json"`? Or is this ready to merge now?

**Comment by alessandrococco** [comment]
> Did you want to add `settingsFile: "ffcniftya.json"`? Or is this ready to merge now?

I would merge now. I'll open a new pull request when I'm done 😄 

---

Thank you @rigrig for the review (and the awesome **ClockFace** lib!)

**Pull request merged**